integration/travis/run_integration.sh

```diff
@@ -35,32 +35,46 @@ ${PROJECT_DIR}/../travis/build_cook_executor.sh
 # Start minimesos
 cd ${PROJECT_DIR}/../travis
 ./minimesos up
+$(./minimesos info | grep MINIMESOS)
+export COOK_ZOOKEEPER="${MINIMESOS_ZOOKEEPER_IP}:2181"
+
+./datomic-free-0.9.5394/bin/transactor $(pwd)/datomic_transactor.properties &
 
 # Start two cook schedulers
 cd ${PROJECT_DIR}/../scheduler
 ## on travis, ports on 172.17.0.1 are bindable from the host OS, and are also
 ## available for processes inside minimesos containers to connect to
 export COOK_EXECUTOR_COMMAND=${COOK_EXECUTOR_COMMAND}
-LIBPROCESS_IP=172.17.0.1 COOK_PORT=12321 COOK_ZOOKEEPER_LOCAL_PORT=3291 COOK_FRAMEWORK_ID=cook-framework-1 lein run ${PROJECT_DIR}/travis/${CONFIG_FILE} &
-LIBPROCESS_IP=172.17.0.1 COOK_PORT=22321 COOK_ZOOKEEPER_LOCAL_PORT=4291 COOK_FRAMEWORK_ID=cook-framework-2 lein run ${PROJECT_DIR}/travis/${CONFIG_FILE} &
+LIBPROCESS_IP=172.17.0.1 COOK_DATOMIC="datomic:free://localhost:4334/cook-jobs" COOK_PORT=12321 COOK_FRAMEWORK_ID=cook-framework-1 COOK_LOGFILE="log/cook-12321.log" lein run ${PROJECT_DIR}/travis/${CONFIG_FILE} &
+LIBPROCESS_IP=172.17.0.1 COOK_DATOMIC="datomic:mem://cook-jobs" COOK_PORT=22321 COOK_ZOOKEEPER_LOCAL=true COOK_ZOOKEEPER_LOCAL_PORT=4291 COOK_FRAMEWORK_ID=cook-framework-2 COOK_LOGFILE="log/cook-22321.log" lein run ${PROJECT_DIR}/travis/${CONFIG_FILE} &
 
 # Wait for the cooks to be listening
 timeout 180s bash -c "wait_for_cook 12321" || curl_error=true
 if [ "$curl_error" = true ]; then
   echo "$(date +%H:%M:%S) Timed out waiting for cook to start listening, displaying cook log"
-  cat ${PROJECT_DIR}/../scheduler/log/cook.log
+  cat ${PROJECT_DIR}/../scheduler/log/cook-12321.log
   exit 1
 fi
-timeout 180s bash -c "wait_for_cook 22321" || curl_error=true
+
+# Start slave after master is running
+LIBPROCESS_IP=172.17.0.1 COOK_DATOMIC="datomic:free://localhost:4334/cook-jobs" COOK_PORT=12322 COOK_FRAMEWORK_ID=cook-framework-1 COOK_LOGFILE="log/cook-12322.log" lein run ${PROJECT_DIR}/travis/${CONFIG_FILE} &
+
+timeout 180s bash -c "wait_for_cook 12322" || curl_error=true
 if [ "$curl_error" = true ]; then
   echo "$(date +%H:%M:%S) Timed out waiting for cook to start listening, displaying cook log"
-  cat ${PROJECT_DIR}/../scheduler/log/cook.log
+  cat ${PROJECT_DIR}/../scheduler/log/cook-12322.log
   exit 1
 fi
+timeout 180s bash -c "wait_for_cook 22321" || curl_error=true
+if [ "$curl_error" = true ]; then
+    echo "$(date +%H:%M:%S) Timed out waiting for cook to start listening, displaying cook log"
+    cat ${PROJECT_DIR}/../scheduler/log/cook-22321.log
+    exit 1
+fi
```

Comment:
Since this section has gotten pretty complicated (3 cooks, 2 connected, 1 separate), I wonder if it would be nice to have a comment somewhere near the top explaining what this is doing and why

Reply:
Done